daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -213,7 +213,7 @@ object SBuiltin {
   final case object SBToQuotedTextParty extends SBuiltin(1) {
     def execute(args: util.ArrayList[SValue], machine: Machine): Unit = {
       val v = args.get(0).asInstanceOf[SParty]
-      machine.ctrl = CtrlValue(SText(s"'${v.value.underlyingString}'"))
+      machine.ctrl = CtrlValue(SText(s"'${v.value}'"))
```

Comment:
To make sure the implied `toString` is benign in the face of future changes:

```suggestion
      machine.ctrl = CtrlValue(SText(s"'${v.value: String}'"))
```

I've discussed with @leo-da doing this as a rule, and replacing `s` in the future with one that only accepts some types for interpolation; for now this is a good stopgap.

Reply:
done